skbio/stats/composition.py

```diff
@@ -838,6 +886,18 @@ def ancom(table, grouping,
     if (table.isnull()).any().any():
         raise ValueError('Cannot handle missing values in `table`.')
 
+    for percentile in percentiles:
+        if not 0.0 <= percentile <= 100.0:
+            raise ValueError('Percentiles must be in the range [0, 100], %r '
+                             'was provided.' % percentile)
+
+    if len(percentiles) != len(set(percentiles)):
```

Comment:
Use `skbio.util.find_duplicates`, check out how it's used [here](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/distance/_base.py#L629). Note that the code is taking multiple passes over `percentiles` at this point, so will fail on an `iterable`.